lora/lora.py

```diff
@@ -220,6 +220,11 @@ def iterate_batches(dset, tokenizer, batch_size, train=False):
 def evaluate(model, dataset, loss, tokenizer, batch_size, num_batches):
     all_losses = []
     ntokens = 0
+
+    # CLI arguments may set num_batches to -1; if so, find the true count
+    if num_batches == -1:
+        num_batches = len(dataset)
```

Comment:
This is not quite right. `len(dataset) != num_batches` since the dataset is not batched yet.

You can do something like what we do in MLX LM:

```python
    index_iterator = iter(range(num_batches)) if num_batches != -1 else iter(int, 1)
```

And then use that in place of `range(num_batches)`.